### PR TITLE
test: fix several bazel compiler-cli tests in windows

### PR DESCRIPTION
**Pull request by alan-agius4**
```
//packages/compiler-cli/integrationtest:integrationtest
//packages/compiler-cli/test/compliance:compliance
```

Partially addresses #29785

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM for dev-infra. 

Also the changes in general look good to me :smile: I did similar stuff in https://github.com/angular/angular/commit/40d64b6b58a41ac7d8e424277647c94d3815db58 to get it work on windows.

File: packages/compiler/test/aot/test_util.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class EmittingCompilerHost implements ts.CompilerHost {
   public effectiveName(fileName: string): string {
     const prefix = '@angular/';
     return angularSourcePath && fileName.startsWith(prefix) ?
-        path.join(angularSourcePath, fileName.substr(prefix.length)) :
+        path.posix.join(angularSourcePath, fileName.substr(prefix.length)) :
```
Comment (gkalpak) [inline packages/compiler/test/aot/test_util.ts]: Why this change? How does it help when `angularSourcePath` intself is a [result of `path.join()`][1]?

[1]: https://github.com/angular/angular/blob/92e2e2cc5ce046d18b2def614c2358a77b0dc66d/packages/compiler/test/aot/test_util.ts#L67

File: packages/compiler/test/aot/test_util.ts [hunk 2]
```diff
@@ -350,7 +350,7 @@ export class MockCompilerHost implements ts.CompilerHost {
     const rxjs = '/rxjs';
     if (name.startsWith('/' + node_modules)) {
       if (nodeModulesPath && name.startsWith('/' + node_modules + rxjs)) {
-        return path.join(nodeModulesPath, name.substr(node_modules.length + 1));
+        return path.posix.join(nodeModulesPath, name.substr(node_modules.length + 1));
```
Comment (gkalpak) [inline packages/compiler/test/aot/test_util.ts]: Why this change? How does it help when `nodeModulesPath` intself is a [result of `path.join()`][1]?

[1]: https://github.com/angular/angular/blob/92e2e2cc5ce046d18b2def614c2358a77b0dc66d/packages/compiler/test/aot/test_util.ts#L66

File: packages/compiler/test/aot/test_util.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/compiler/test/aot/test_util.ts]: You are right

File: packages/compiler/test/aot/test_util.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/compiler/test/aot/test_util.ts]: Right!

**Review by alxhub — APPROVED** [review]
